op-challenger/fault/types.go

```diff
@@ -19,19 +19,31 @@ type TraceProvider interface {
 	Get(i uint64) (common.Hash, error)
 }
 
-type Claim struct {
+// ClaimData is the core of a claim. It must be unique inside a specific game.
+type ClaimData struct {
 	Value common.Hash
 	Position
 }
 
-type Response struct {
-	Attack bool // note: can we flip this to true == going right / defending??
-	Value  common.Hash
-	Parent Claim
+// Claim extends ClaimData with information about the relationship between two claims.
+// It uses ClaimData to break cyclicity without using pointers.
+// If the position of the game is Depth 0, IndexAtDepth 0 it is the root claim
+// and the Parent field is empty & meaningless.
+type Claim struct {
+	ClaimData
+	Parent ClaimData
+}
+
+func (c *Claim) IsRoot() bool {
+	return c.Position.IsRootPosition()
+}
+
+func (c *Claim) DefendsParent() bool {
+	return c.IndexAtDepth() != c.Parent.IndexAtDepth()
 }
 
 // Responder takes a response action & executes.
 // For full op-challenger this means executing the transaction on chain.
 type Responder interface {
-	Respond(ctx context.Context, response Response) error
+	Respond(ctx context.Context, response Claim) error
```

Comment:
We lose the explicit `Response.Attack` information by doing this, so the solver `NextMove` just spits out a `Claim`. Is that all we should need for the game solver (in particular `Agent`) to respond?

Reply:
Yea - the claim type now has `IsDefend` available.

Reply:
Got it yes that makes sense!